contracts/token/ERC1155/ERC1155.sol

```diff
@@ -311,17 +332,53 @@ contract ERC1155 is Context, ERC165, IERC1155, IERC1155MetadataURI {
         require(account != address(0), "ERC1155: attempting to burn batch of tokens on zero account");
         require(ids.length == values.length, "ERC1155: burnt IDs and values must have same lengths");
 
-        for(uint i = 0; i < ids.length; i++) {
+        address operator = _msgSender();
+
+        _beforeTokenTransfer(operator, account, address(0), ids, values, "");
+
+        for (uint i = 0; i < ids.length; i++) {
             _balances[ids[i]][account] = _balances[ids[i]][account].sub(
                 values[i],
                 "ERC1155: attempting to burn more than balance for some token"
             );
         }
 
-        emit TransferBatch(_msgSender(), account, address(0), ids, values);
+        emit TransferBatch(operator, account, address(0), ids, values);
     }
 
+    /**
+     * @dev Hook that is called before any token transfer. This includes minting
+     * and burning, as well as batched variants.
+     *
+     * The same hook is called on both single and batched variants. For single
+     * transfers, the length of the `id` and `amount` arrays will be 1.
+     *
+     * Calling conditions (for each `id` and `amount` pair):
+     *
+     * - when `from` and `to` are both non-zero, `amount` of ``from``'s tokens
```

Comment:
This list is not showing up as a list in the rendered docs.

Reply:
Are you sure? It seems fine here: https://deploy-preview-2267--openzeppelin-contracts-docs.netlify.app/contracts/3.x/api/token/erc1155#ERC1155-_beforeTokenTransfer-address-address-address-uint256---uint256---bytes-

Reply:
I commented on the wrong one. See https://deploy-preview-2267--openzeppelin-contracts-docs.netlify.app/contracts/3.x/api/token/erc1155#IERC1155-safeTransferFrom-address-address-uint256-uint256-bytes-

Reply:
Ah good catch